lib/logstash/codecs/auto_flush.rb

```diff
@@ -11,16 +11,21 @@ def start
     # can't start if pipeline is stopping
     return self if stopped?
     if pending?
-      @task.reset
+      @task.cancel
```

Comment:
Nevermind, the cancel will work in that case, making the task finish and never run it again.